src/toil/leader.py

```diff
@@ -77,8 +77,13 @@ def callback(fileHandle):
             except AttributeError:
                 pass
             else:
-                for log in logs:
-                    logger.info("%s:    %s", log.jobStoreID, log.text)
+                jobStoreIDs = {log.jobStoreID for log in logs}
+                # we may have multiple jobs per worker
+                for jobStoreID in jobStoreIDs:
+                    jobLogs = [log.text for log in filter(lambda log: log.jobStoreID == jobStoreID, logs)]
```

Comment:
On second thought, is there a way to make this O(n) rather than O(n*m)? n being number of logs, m being the number of distinct jobs.